src/linux_serial_ccsds/linux_serial_ccsds.h

```diff
@@ -0,0 +1,160 @@
+ * @brief Structure for driver internal data.
+ *
+ * This structure is allocated by runtime and the pointer is passed to all driver functions.
+ * The name of this structure shall match driver definition from ocarina_components.aadl
+ * and has suffix '_private_data'.
+ */
+class linux_serial_ccsds_private_data final
+{
+  public:
+    /**
+     * @brief  Constructor.
+     *
+     * Construct empty object, which needs to be initialized using @link linux_serial_ccsds_private_data::init
+     * before usage.
+     */
+
+    linux_serial_ccsds_private_data();
+
+    /**
+     * @brief  Destructor.
+     *
+     * Destruct created object
+     */
+    ~linux_serial_ccsds_private_data();
+
+    /**
+     * @brief Initialize driver.
+     *
+     * Driver needs to be initialized before start.
+     *
+     * @param bus_id         Identifier of the bus, which is used by driver
+     * @param device_id      Identifier of the device
+     * @param device_configuration Configuration of device
+     * @param remote_device_configuration Configuration of remote device
+     */
+    void driver_init(const SystemBus bus_id,
+                     const SystemDevice device_id,
+                     const Serial_CCSDS_Linux_Conf_T* const device_configuration,
+                     const Serial_CCSDS_Linux_Conf_T* const remote_device_configuration);
+    /**
+     * @brief Receive data from remote partitions.
+     *
+     * This function receives data from remote partition and sends it to the Broker.
+     */
+    void driver_poll();
+    /**
+     * @brief send data to remote partition.
+     *
+     * @param data           The Buffer which data to send to connected remote partition
+     * @param length         The size of the buffer
+     */
+    void driver_send(const uint8_t* data, const size_t length);
+
+  private:
+    static constexpr int DRIVER_THREAD_PRIORITY = 1;
+    static constexpr int DRIVER_THREAD_STACK_SIZE = 65536;
+    static constexpr size_t DRIVER_RECV_BUFFER_SIZE = 8 * 1024;
+
+    void driver_init_baudrate(const Serial_CCSDS_Linux_Conf_T* const device, int* cflags);
+    void driver_init_character_size(const Serial_CCSDS_Linux_Conf_T* const device, int* cflags);
+    void driver_init_parity(const Serial_CCSDS_Linux_Conf_T* const device, int* cflags);
+
+    int serialFd{ -1 };
+    enum SystemBus m_serial_device_bus_id;
+    enum SystemDevice m_serial_device_id;
+    const Serial_CCSDS_Linux_Conf_T* m_serial_device_configuration{};
+    const Serial_CCSDS_Linux_Conf_T* m_serial_remote_device_configuration{};
+    taste::Thread m_thread;
+
+    uint8_t m_recv_buffer[DRIVER_RECV_BUFFER_SIZE];
+    Escaper escaper{};
+};
+
+namespace taste {
+
+/**
+ * @brief Function which implements receiving data from remote partition.
+ *
+ * Functions works in separate thread, which is initialized by @link LinuxSerialCcsdsSend
+ *
+ * @param private_data   Driver private data, allocated by runtime
+ */
+void LinuxSerialCcsdsPoll(void* private_data);
```

Comment:
This in turn probably should be underscore separated

Reply:
Rafał created those names in a similar manner for linux_ip_socet. I just copied the convention. He could have a good reason for it. @rphh

Reply:
Name enforced by definition from ocarina_components.aadl